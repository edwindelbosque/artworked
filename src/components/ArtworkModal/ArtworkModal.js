import React from 'react';
import './ArtworkModal.scss';
import Nav from '../Nav/Nav';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { addFavorite, removeFavorite } from '../../actions/index';
import { getAlbumTracks } from '../../util/apiCalls';

const Artwork = ({
	result,
	favorites,
	addFavorite,
	removeFavorite,
	isFavorites
}) => {
	const { name, artist, releaseYear, hqArtwork, id } = result;

	const getSongs = async () => {
		await getAlbumTracks(id);
	};

	const handleFavorite = () => {
		getSongs();
		favorites.find(favorite => favorite.id === id)
			? removeFavorite(id)
			: addFavorite(result);
	};

	const heartToggle = favorites.find(favorite => favorite.id === id)
		? 'is-active'
		: '';

	const routeHandler = isFavorites ? 'favorites' : '';

	return (
		<>
			<Link to={`/${routeHandler}`}>
				<div className='modal-backdrop'></div>
			</Link>
			<Nav />
			<article className='ArtworkModal modalContainer'>
				<img alt={`${name} album artwork`} src={hqArtwork} />
				<div>
					<h3 className='title'>{name}</h3>
					<h3 className='artist'>{artist}</h3>
					<h3 className='year'>{releaseYear}</h3>
					<a
						href={hqArtwork}
						target='_blank'
						rel='noopener noreferrer'
						download>
						<button className='artwork-button'>Get Artwork</button>
					</a>
					<div
						className={`heart ${heartToggle}`}
						onClick={handleFavorite}></div>
				</div>
			</article>
		</>
	);
};

const mapStateToProps = ({ favorites, isFavorites }) => ({
	favorites,
	isFavorites
});

const mapDispatchToProps = dispatch => {
	return bindActionCreators(
		{
			addFavorite,
			removeFavorite
		},
		dispatch
	);
};

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(Artwork);
