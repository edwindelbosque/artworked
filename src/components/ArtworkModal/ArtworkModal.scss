.modalContainer {
	position: fixed;
	z-index: 100;
	top: 50%;
	left: 50%;
	transform: translate3d(-50%, -50%, 0);
	display: flex;
	justify-content: space-between;
	margin: auto;
	background: rgb(75, 75, 75);
	margin-top: 20px;
	border-radius: 4px;
	min-width: 20rem;
	max-width: 60rem;
	height: 578px;
	border: 1px solid gray;
}

.modal-backdrop {
	width: 100vw;
	height: 110vh;
	position: fixed;
	top: 50%;
	left: 50%;
	background: rgba(0, 0, 0, 0.63);
	background-position: center top;
	background-size: cover;
	backdrop-filter: blur(8px);
	transform: translate3d(-50%, -50%, 0);
	object-fit: cover;
	z-index: 1;
}

.ArtworkModal {
	img {
		height: 576px;
		border-radius: 4px 0 0 4px;
		box-shadow: 3px 0px 10px rgba(0, 0, 0, 0.384);
	}

	div {
		margin-left: 20px;
		margin-right: 20px;
		width: 350px;
		display: flex;
		flex-direction: column;

		.heart {
			width: 100px;
			position: absolute;
			right: -1px;
			top: 123px;
			height: 100px;
			background: url('https://cssanimation.rocks/images/posts/steps/heart.png')
				no-repeat;
			background-position: 0 0;
			cursor: pointer;
			transition: background-position 1s steps(28);
			transition-duration: 0s;

			&.is-active {
				transition-duration: 1s;
				background-position: -2800px 0;
			}
		}

		h3 {
			background: rgb(192, 192, 192);
			color: rgb(43, 43, 43);
			padding: 5px 10px;
		}
	}
	.title {
		margin-top: 20px;
		font-family: Montserrat;
		padding-top: 12px;
		font-weight: 600;
		padding-bottom: 12px;
		background: rgb(228, 228, 228);
		border-radius: 4px 4px 0 0;
		border-bottom: none;
	}

	.artist {
		background: rgb(214, 214, 214);
		padding-bottom: 6px;
		font-weight: 500;
		padding-top: 6px;
	}

	.year {
		border-radius: 0 0 4px 4px;
		font-size: 14px;
		background: rgb(192, 192, 192);
		border-top: none;
		font-weight: 500;
		padding-bottom: 6px;
		padding-top: 6px;
		box-shadow: 0 3px 5px rgba(0, 0, 0, 0.226);
	}
}

.artwork-button {
	border: 1px solid rgb(0, 255, 251);
	color: rgb(233, 254, 255);
	border-radius: 2px;
	font-weight: 400;
	width: 70%;
	font-size: 20px;
	cursor: pointer;
	padding: 10px 32px;
	background: rgba(0, 0, 0, 0.098);
	margin-top: 20px;
	box-shadow: 0 2px 3px rgba(0, 0, 0, 0.242);
	transition: 0.05s;

	&:focus {
		outline: none;
	}

	&:hover {
		box-shadow: 0 3px 4px rgba(0, 0, 0, 0.26);
		background: rgba(186, 186, 186, 0.064);
		transition: 0.05s;
	}
}

a {
	width: 100%;
}
