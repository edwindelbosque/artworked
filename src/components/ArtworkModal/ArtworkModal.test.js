import React from 'react';
import { ArtworkModal } from './ArtworkModal';
import { shallow } from 'enzyme';

describe('ArtworkModal', () => {
	let wrapper;
	const result = { information: 'yes' };
	const favorites = [{}, {}, {}];
	const match = { params: ['path'] };

	beforeEach(() => {
		wrapper = shallow(
			<ArtworkModal result={result} favorites={favorites} match={match} />
		);
	});

	it('should match ArtworkModal snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
