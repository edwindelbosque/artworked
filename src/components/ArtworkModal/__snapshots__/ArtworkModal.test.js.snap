// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtworkModal should match ArtworkModal snapshot 1`] = `
<Fragment>
  <Link
    to="/"
  >
    <div
      className="modal-backdrop"
    />
  </Link>
  <Connect(Nav) />
  <article
    className="ArtworkModal modalContainer"
  >
    <img
      alt="undefined album artwork"
    />
    <div>
      <h3
        className="title"
      />
      <h3
        className="artist"
      />
      <h3
        className="year"
      />
      <a
        download={true}
        rel="noopener noreferrer"
        target="_blank"
      >
        <button
          className="artwork-button"
        >
          Get Artwork
        </button>
      </a>
      <div
        className="heart is-active"
        onClick={[Function]}
      />
    </div>
  </article>
</Fragment>
`;
